karapace/rapu.py

```diff
@@ -286,6 +286,15 @@ async def _handle_request(
                     charset = options.get("charset", "utf-8")
                     body_string = body.decode(charset)
                     rapu_request.json = json_decode(body_string)
+
+                    # Prevent string, int etc. going further from here
+                    if not isinstance(rapu_request.json, dict):
+                        http_error(
+                            message="Malformed request",
+                            content_type=JSON_CONTENT_TYPE,
+                            code=HTTPStatus.BAD_REQUEST,
+                        )
```

Comment:
issue: This looks like it should not be covered by the try-block, but should rather be appended after it?

Reply:
The try-block catches `UnicodeDecodeError`, `LookupError`, and `ValueError`, but not `HTTPResponse`.

The issue lies in maintainability: adding code to a try-block that doesn't belong there can make it significantly harder for a reader of the code to distinguish what possible paths through the code there are. As the opposite of having the minimal amount of code being covered makes it very clear, and there are no advantages of adding unrelated code to a try-block, I don't see this as opinionated.

Reply:
Sounds reasonable, fixed now.